### ISSUE TITLE
Email accounts colors

### ISSUE THREAD

**Issue by joaojotta** [nextcloud/mail#486]
So, this doesn't need a lot to understand.

Issue/Request, there's simply no way to customize colors on different email accounts on the Mail app. It's just setup randomly by default (or generated by the email account name).
This caused me to have all 3 accounts with the same color.

Any way to add this?

Thank you!

**Comment by ChristophWurst** [comment]
@skjnldsv any ideas? :)

**Comment by jancborchardt** [comment]
We could add a picker, just like in the calendar? What do you use there @georgehrke @tcitworld @raghunayyar?

**Comment by georgehrke** [comment]
We don't use any special library. It's just 8 colored boxes with an additional input for HTML 5 Input color

**Comment by tcitworld** [comment]
Native color picker or random color button as default if the native color picker is not available.

**Comment by skjnldsv** [comment]
Agree with the calendar way! :)

**Comment by joaojotta** [comment]
Choosing a color from existing 8 or customizing would be amazing! Whatever is easier.

**Comment by alexanderdd** [comment]
+1

**Comment by jancborchardt** [comment]
Recently I also ran into this, where my 2 (!) accounts have the same color. We could do it similar to the Calendar where a newly created calendar picks a color which has not been used. So if the autogenerated color is too close to one which already exists, we simply move along a bit on the color wheel?

**Comment by jancborchardt** [comment]
@alexanderdd please note that this app is open source, so you can help in several ways making this happen: Contribute code, open a bounty for it on [Bountysource](https://www.bountysource.com/), spread the word to get developers to implement this, etc etc.

We discuss and decide what makes sense regarding design and development in a step-by-step process:
1. First step is to not have the colors of 2 accounts conflict. If you have only 2 accounts which happen to get the same colors through the algorithm, you shouldn’t need to fix it yourself via a color picker. Most people don’t change defaults, so we need this anyway.
2. In the future, and if possibly people help out, we can have a color picker there.

**Comment by alexanderdd** [comment]
Ok, thanks for this response. Now that you say that people don't change
defaults, I agree that there should be an algorithm to avoid same colors
for accounts. Before this explanation, I thought: why spend work time on
the algorithm when multiple people have already expressed that they want a
color picker? That's why I posted the comment above.

(Offtopic:)
Thank you also for the ideas how to contribute. I can not code myself, but
I am reporting bugs and I follow the bugs that are important to me as a
user. I already posted bounties on other bugs that are more important to
me, I don't want to post a bounty on this one. To be honest, I don't know
where to spread the word to attract developers.

Am Fr., 30. Nov. 2018 um 17:53 Uhr schrieb Jan-Christoph Borchardt <
notifications@github.com>:

**Comment by xMartin** [comment]
How about defining, say, 12 fixed colors and just assigning one each account from the top? Too boring?

**Comment by jancborchardt** [comment]
@xMartin yup, that’s definitely a simple and good first approach – basically what I meant above at https://github.com/nextcloud/mail/issues/486#issuecomment-440646864 :)

**Comment by xMartin** [comment]
@jancborchardt yeah, but that implies calculating difference between colors. Could you just define 8-12 color codes as a spec and then the account always have the same color counting up.

**Comment by jancborchardt** [comment]
@georgehrke @skjnldsv where do we have this color list that we use in Calendar and/or in general?

**Comment by georgehrke** [comment]
Calendar just generates a random color and doesn't check for color distance at the moment

**Comment by georgehrke** [comment]
We have some default colors in the Nextcloud-vue color picker:
https://github.com/nextcloud/nextcloud-vue/blob/master/src/components/ColorPicker/ColorPicker.vue#L180

**Comment by skjnldsv** [comment]
> @georgehrke @skjnldsv where do we have this color list that we use in Calendar and/or in general?

same, we use a hash and let it do its randomness :)
We have 18 colours iirc, so it's one chance out of 18 to get the same color again :see_no_evil:  (roughtly, my algorithm was rather good in spreading the data)

**Comment by ChristophWurst** [comment]
Hey there,

I'd just like to let you know that due to the [ongoing issue with Bountysource](https://help.nextcloud.com/t/dont-use-bountysource-anymore/84943) we're discussing ways to move to a different platform. As it looks right now, Bountysource might claim some of the money that was contributed by the community. We will try our best to prevent that. Historically we – the Nextcloud GmbH employees – simply put the claimed bounties back to other tickets until one was solved by a community member. So the money was always *from the community* and *for the community*. We'll therefore move to a better platform and currently don't post any new bounties until this situation is resolved. The current goal is to move/donate the money to [Nextcloud include](https://nextcloud.com/include/) but details will follow in the [forum post](https://help.nextcloud.com/t/dont-use-bountysource-anymore/84943).

Hope this makes sense. If you have any questions please ask them on the forum. This comment will be copy-pasted to all open issues with bounties.

Christoph and the Mail team

**Comment by trougnouf** [comment]
Is there a way to define the color in SQL or some config file? I haven't found the variable so far.

**Comment by ChristophWurst** [comment]
There is none, hence the ticket.
